### DISCUSSION TITLE
Error reading landscape ASCII haeders - aborting

### DISCUSSION THREAD

**Discussion by mariesoret** [RangeShifter/RangeshiftR-tutorials#37] [Q&A] [answered]
Hi, 
I'm trying to use rangeshiftR but the simulation stop before opening asc file with the following message : 

```
*** Format error in header line of AO_EXP/Inputs/landscape.asc: 1 error detected
*** Format error in header line of AO_EXP/Inputs/AO_COREHAB_ID.asc: 1 error detected
*** Format error in header line of AO_EXP/Inputs/AO_CF_1_100.asc: 1 error detected
Error reading landscape ASCII haeders - aborting
```

I checked the headers of my ascii files, i really don't see the problem with it. For example here is the headers of landscape.asc:
```
NCOLS 2262
NROWS 1409
XLLCORNER 529011.893600
YLLCORNER 6445115.312600
CELLSIZE 10.000000
NODATA_VALUE -9999
```

Could this be a problem with the size of the file ? 
Anyone had this problem before me ?

**Comment by JetteReeg** [comment]
Hi mariesoret,
I also can't see a problem with the header at first glance. I don't think that it is a matter of the size of the file. But have you tried using a smaller landscape file?
Could you maybe provide me with a minimal working example so I could run some tests?

**Comment by mariesoret** [comment]
Hi, thanks for answering here is a zip with RS directory and Rmd file:
 
[C-RANGESHIFTER-PAUSE.zip](https://github.com/RangeShifter/RangeshiftR-tutorials/files/10121518/C-RANGESHIFTER-PAUSE.zip)

I will try on a smaller area

**Comment by JetteReeg** [comment, marked as answer]
Hi,
I was able to solve the error after I loaded the raster files in R and saved them again:

```
libary(raster)
library(rgdal)   
landscape <- raster("AO_EXP/Inputs/landscape.asc")   
writeRaster(landscape,"AO_EXP/Inputs/landscape.asc", format="ascii")
```

So it seems that there are some (hidden) format errors in the files.

A small side note: I saw that your habitat codes start with 0. Please check the help file of ImportedLandscape():

> Raster with habitat codes (HabPercent=FALSE)
> In this option each habitat or land-cover type has a unique integer code. Each cell in the file contains a single habitat code and 100 percent coverage is assumed for the cell. The landscape is therefore composed of discrete habitat cells. The codes are required to be sequential integers **starting from 1 and ranging to Nhabitats**.

**Reply by mariesoret** [reply]
Thanks !

**Comment by dsbennett2023** [comment]
Sorry to hijack a thread, but I've been having the same problem. 

Have tried importing and then re-exporting in R like you did above, but it still doesnt work. Keep getting the "format error in header lines of the selected landscape file" (when trying with txt in GUI)  or "*** Format error in header line " when trying with ASCII files in RangeshiftR. 

Have attached examples in the zip. 

The files are very big - over 250mb when uncompressed. Could it be that they are too big for the software to read?

Reducing file size by using a coarser resolution has not worked yet though. 

[landscape4inputs.zip](https://github.com/RangeShifter/RangeshiftR-tutorials/files/11111810/landscape4inputs.zip)

Sorry to bother you

**Comment by JetteReeg** [comment]
Hi,
I tested the landscape file you provided with a minimal RS setup and got 3 errors while reading the header line.
Similar to the solution aove, I loaded the file in R and write the raster again, but added the datatype and NAflag:

```library(raster)
library(RangeShiftR)
landscape <- raster("landscape4arc.asc")
writeRaster(landscape,
            filename = "landscape4arc.asc",
            format = "ascii",
            overwrite = TRUE, 
           datatype = 'INT1S', 
           NAflag = -999)
```

With this I could solve only 2 errors. So I checked the header manually and saw that the resolution was given as decimal number (25.00001). I set it back to 25. After that no error messages occured reading the LandscapeFile headers.

However, the provided landscape file is indeed quite large and I actually interrupted the RangeShiftR test simulation while it was reading in the landscape file because it took too long. Nevertheless, RangeShiftR should be able to run with such a big landscape, but you need to be aware that it might need a lot of resources and the run time could get quite long. If you want to stick to that landscape, I would suggest to try running it on a High Performance Cluster.

**Reply by JetteReeg** [reply]
Hi,

There seem to be two problems:
1. : RangeshiftR is not able to open the PatchFile. Have you checked whether you can open the file outside of RangeShiftR? (e.g. via terra::rast(<filepath>)?
2. There seem to be a problem in the headers of the landscape file.

I have tried a simple example using the patch file you provided. As a landscape file I transformed the patch file into a landscape file with habitat codes 1 and 2. I was not able to reproduce your error; both landscape and patch files could be loaded.

**Reply by JetteReeg** [reply]
Hi, 

sorry for the delayed response.
If you generate the artificial landscape outside of RangeShiftR, you can have an artificial landscape with habitat codes/types 1 (=matrix) and 2 (=habitat) together with a patch-based model. 

In general:
If you have a patch file ready and would like to generate the landscape file based on it, you should set all values with patch ID == 0 to 1 (=matrix) and all values with patch ID >0 to 2 (=habitat).

Example if you have loaded the patch file as raster object _patchmap_ in R:
```
landmap <- patchmap # copy object to generate landscape map
values(landmap)[values(landmap)>0] <- 2 # habitat
values(landmap)[values(landmap)==0] <- 1 # matrix
terra::writeRaster(landmap, "Model/Inputs/Landscape.asc", NAflag = -999, datatype = "INT2S", overwrite = T) # you need to check whether cellsize parameter is an integer values and not a float
```

Regarding to your question about the patch file: If you are able to plot the patch file in R, it should be fine for RangeShiftR. Could it be, that all matrix cells are NAs instead of (patch ID) 0? You could use a similar code as I posted above to change patch ID values. 
Again, after saving the .asc file, you should check the header (cellsize needs to be an integer value).
